src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -194,6 +195,21 @@ private void ChangeProjectProperty<T>(ref T field, T newValue, Func<Solution, So
 
                 field = newValue;
 
+                if (logThrowAwayTelemetry)
+                {
+                    var telemetry = _workspace.Services.GetRequiredService<IWorkspaceTelemetryService>();
+                    // If we already have a compilation, then this change will mean it gets throw away, so log it
+                    if (_workspace.CurrentSolution.State.TryGetCompilation(Id, out var previousCompilation))
```

Comment:
Do you want to count trees by instead looping through the documents and seeing which ones have trees that way? If we've computed some trees but not all of them, we won't have built a compilation yet but are still throwing away (non-trivial) work.